plugins/redirect/gatsby-node.js

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-restricted-syntax */
+const path = require('path');
+const fs = require('node:fs/promises');
+
+const getMetaRedirect = require('./redirect');
+
+async function writeRedirectsFile(redirects, folder, pathPrefix) {
+  if (!redirects.length) return;
+
+  for (const redirect of redirects) {
+    const { fromPath, toPath } = redirect;
+
+    const FILE_PATH = path.join(
+      folder,
+      fromPath.replace(pathPrefix, ''),
+      'index.html'
+    );
+
+    try {
+      const data = getMetaRedirect(toPath);
+      await fs.writeFile(FILE_PATH, data);
```

Comment:
I don't think we should use await here. Otherwise, this slow downs the build. Right now, hundreds of redirect files are generated. Gatsby API allows callbacks and asynchronous work https://www.gatsbyjs.com/docs/reference/config-files/gatsby-node/#async-vs-sync-work.

We could parallel the creation of those files. We're using the `async` npm package which is quite neat: https://caolan.github.io/async/v3/docs.html#parallel